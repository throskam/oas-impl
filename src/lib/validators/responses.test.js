import createResponsesValidator from './responses'

describe('Empty', () => {
  const response = {}

  const validator = createResponsesValidator(response)

  it('should return no errors', () => {
    expect(validator()).toMatchSnapshot()
  })
})

describe('Exact', () => {
  const responses = {
    200: {
      content: {
        'application/json': {
          schema: {
            type: 'integer'
          }
        }
      }
    }
  }

  const validator = createResponsesValidator(responses)

  it('should return no errors when the value is valid', () => {
    const payload = { content: 1, mediaType: 'application/json', status: 200 }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return a validation error when the value is invalid', () => {
    const payload = { content: 'string', mediaType: 'application/json', status: 200 }
    expect(validator(payload)).toMatchSnapshot()
  })
})

describe('Default', () => {
  const responses = {
    default: {
      content: {
        'application/json': {
          schema: {
            type: 'integer'
          }
        }
      }
    }
  }

  const validator = createResponsesValidator(responses)

  it('should return no errors when the value is valid', () => {
    const payload = { content: 1, mediaType: 'application/json', status: 200 }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return a validation error when the value is invalid', () => {
    const payload = { content: 'string', mediaType: 'application/json', status: 200 }
    expect(validator(payload)).toMatchSnapshot()
  })
})

describe('Wildcard', () => {
  const responses = {
    '2XX': {
      content: {
        'application/json': {
          schema: {
            type: 'integer'
          }
        }
      }
    }
  }

  const validator = createResponsesValidator(responses)

  it('should return no errors when the value is defined', () => {
    const payload = { content: 1, mediaType: 'application/json', status: 200 }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return a validation error when the value is valid', () => {
    const payload = { content: 'string', mediaType: 'application/json', status: 200 }
    expect(validator(payload)).toMatchSnapshot()
  })
})
