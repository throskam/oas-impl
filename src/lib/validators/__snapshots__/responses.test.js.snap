// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default should return a validation error when the value is invalid 1`] = `
Array [
  Object {
    "ajv": Object {
      "instancePath": "",
      "keyword": "type",
      "message": "must be integer",
      "params": Object {
        "type": "integer",
      },
      "schemaPath": "#/type",
    },
    "path": "responses[200].response.content[application/json].schema",
    "rule": "schema-invalid",
    "value": "string",
  },
]
`;

exports[`Default should return no errors when the value is valid 1`] = `Array []`;

exports[`Empty should return no errors 1`] = `Array []`;

exports[`Exact should return a validation error when the value is invalid 1`] = `
Array [
  Object {
    "ajv": Object {
      "instancePath": "",
      "keyword": "type",
      "message": "must be integer",
      "params": Object {
        "type": "integer",
      },
      "schemaPath": "#/type",
    },
    "path": "responses[200].response.content[application/json].schema",
    "rule": "schema-invalid",
    "value": "string",
  },
]
`;

exports[`Exact should return no errors when the value is valid 1`] = `Array []`;

exports[`Wildcard should return a validation error when the value is valid 1`] = `
Array [
  Object {
    "ajv": Object {
      "instancePath": "",
      "keyword": "type",
      "message": "must be integer",
      "params": Object {
        "type": "integer",
      },
      "schemaPath": "#/type",
    },
    "path": "responses[200].response.content[application/json].schema",
    "rule": "schema-invalid",
    "value": "string",
  },
]
`;

exports[`Wildcard should return no errors when the value is defined 1`] = `Array []`;
